pkg/forth/forth.go

```diff
@@ -265,10 +302,15 @@ func hostname(f Forth) {
 }
 
 func hostbase(f Forth) {
-	host := f.Pop()
+	host := String(f)
 	f.Push(strings.TrimLeft(host, "abcdefghijklmnopqrstuvwxyz -"))
 }
 
+// NewWord allows for definition of new operators from strings.
+// For example, should you wish to create a word which adds a number
+// to itself twice, you can call:
+// NewWord(f, "d3d", "dup dup + +")
+// which does two dups, and two adds.
 func NewWord(f Forth, name, command string) {
 	newword := func(f Forth) {
 		iEval(f, command)
```

Comment:
at line 319 in the new version:
         opmap[name] = newword
should that use the NewOp interface function?

(i couldn't comment on it directly, or at least it wasn't clear how)

Reply:
I went back and forth on that question and went with the direct assignment.